### ISSUE TITLE
[Typescript Client] Lists and non null

### ISSUE THREAD

**Issue by Nebulis** [dotansimha/graphql-code-generator#1223]
**Describe the bug**

Generated code by typescript client plugin doesn't respect nullable list / elements:

| Graphql Type  | Expected Typescript Type | Actual Generated Type |
| ------------- | -------------------------- | -------------   |
| [String]      | `Maybe<Array<Maybe<string>>>` | `Maybe<string[]>` |
| [String]!     | `Array<Maybe<string>>`        | `string[]`        |
| [String!]     | `Maybe<string[]>`            | `Maybe<string[]>` |
| [String!]!    | `string[]`                   | `string[]`       |

1. Schema: (change list accordingly) 
```graphql
type CountriesPayload {
 countries: [String]
}
type Query {
 countries: CountriesPayload!
}
```
2. Query
```javascript
gql`
  query Countries {
    countries {
      countries
    }
  }
`
```
3. Run tests from typescript-client plugin (I also reproduced with gql-gen from a repo)

**Environment:**
 - OS: linux
 - Codegen: 0.15.2
 - Node: 10.15.0

**Additional context**
#169 and #260 are related to the same problem but are fixed (probably issue on different components / plugins). I hope I didn't miss anything

**Comment by Nebulis** [comment]
Turns out the bug only happened when using `noNamespaces` option. There is a specific use case for arrays and noNamespaces (which is not covered by tests).

I open an MR (#1265) to remove this check, I'm not sure of the impact of this. Happy to discuss about it :)

**Comment by dotansimha** [comment]
Available in 0.17.0, thanks @Nebulis !
